### ISSUE TITLE
[Feature Request] Reindex should use each shard to perform slice tasks instead of coordination node

### ISSUE THREAD

**Issue by kkewwei** [opensearch-project/OpenSearch#14516]
### Is your feature request related to a problem? Please describe

In reindex, coordination node will create multi threads to process each slice(contains search, cleaning data, bulk writing), there are two problems:
1. reindex will occupy a lot of resources of the coordination node.
2. coordination node process multi slices, it's low efficiency.

### Describe the solution you'd like

If each shard do the slice task? just like snapshot restore.  include delete_by_query, update_by_query.

### Related component

Indexing:Performance

**Comment by kkewwei** [comment]
@ankitkala please help confirm it.

**Comment by ankitkala** [comment]
Hi,  you should be able to verify this by running multiple node cluster locally using [gradle run](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#run-opensearch) and test reindex with slice.

**Comment by kkewwei** [comment]
@ankitkala, no, now the reindex slices are executed on the node(coordination or data node) who first receive the reindex reuqest.

**Comment by vikasvb90** [comment]
Thanks @kkewwei ! Looking forward to have a PR for the same along with some improved benchmark numbers.
